### PR TITLE
Update Docker setup instructions to discard setup container

### PR DESCRIPTION
**Pull request by mmahacek**
## Proposed commit message

Add --rm to the setup command to automatically remove the container used to setup. This prevents leaving behind a single-use container instance.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [X] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

No, docs only

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -136,6 +136,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Enable early event encoding in the Elasticsearch output, improving cpu and memory use {pull}38572[38572]
 - The environment variable `BEATS_ADD_CLOUD_METADATA_PROVIDERS` overrides configured/default `add_cloud_metadata` providers {pull}38669[38669]
 - Update to Go 1.22.7. {pull}41018[41018]
+- Update container install docs to remove setup container automatically on completion. {pull}41069[41069]
```
Comment (belimawr) [inline CHANGELOG.next.asciidoc:139]:
```suggestion
```

We don't need changelog entries for docs update, the changelog goal is to inform users of new features/changes they need to be aware when updating.

**Review by belimawr — CHANGES_REQUESTED** [review]
@mmahacek could you also update this PR to target `main`? We can backport it to 8.15 if needed.

**Comment by mmahacek** [comment]
AAAAH!  I need to update the base branch!

**Comment by mmahacek** [comment]
I need to start over here...

**Pull request closed without merging**